modules/videoio/src/cap_ffmpeg.cpp

```diff
@@ -125,7 +125,7 @@ class icvInitFFMPEG
             {
                 wchar_t path[MAX_PATH];
                 size_t sz = GetModuleFileNameW(m, path, sizeof(path));
-                if (sz > 0 && ERROR_SUCCESS == GetLastError())
+                if (sz > 0)
```

Comment:
[MSDN](https://msdn.microsoft.com/en-us/library/windows/desktop/ms683197(v=vs.85).aspx) states this:
> If the buffer is too small to hold the module name, the string is truncated to nSize characters including the terminating null character, the function returns nSize, and the function sets the last error to ERROR_INSUFFICIENT_BUFFER.

Perhaps check should be:
```
if (sz > 0 && sz < MAX_PATH)
```

but there are additional notes (including Windows XP) ...

So could you try to force reset of error code before function call in your configuration:
```
SetLastError(ERROR_SUCCESS);
```

BTW, Line above is buggy too:
```.diff
-size_t sz = GetModuleFileNameW(m, path, sizeof(path));
+size_t sz = GetModuleFileNameW(m, path, MAX_PATH);
```

because `sizeof(path)` is in bytes.

Also there is other important thing for Windows XP:
> Windows XP:  The string is truncated to nSize characters and is not null-terminated.

so we need to add `path[sz] = 0` into "if" body.